"""Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT."""

from __future__ import annotations
import dataclasses
from ..shared import error_code_enum as shared_error_code_enum
from ..shared import error_source_enum as shared_error_source_enum
from ..shared import error_type_enum as shared_error_type_enum
from dataclasses_json import Undefined, dataclass_json
from sdk import utils


@dataclass_json(undefined=Undefined.EXCLUDE)
@dataclasses.dataclass
class Error:
    r"""The error structure that gets returned with almost all failed API calls"""
    
    error_code: shared_error_code_enum.ErrorCodeEnum = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.get_field_name('error_code') }})  
    error_source: shared_error_source_enum.ErrorSourceEnum = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.get_field_name('error_source') }})  
    error_type: shared_error_type_enum.ErrorTypeEnum = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.get_field_name('error_type') }})  
    message: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.get_field_name('message') }})
    r"""An error message associated with the failed API call"""  
    